### PR TITLE
Fixes #19926: Support reporting logic metadata in language

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/19926

### CODE REVIEW

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
PR updated with a new commit

File: language/src/technique.rs [hunk 1]
```diff
@@ -200,8 +202,36 @@ enum MethodElem {
     MethodBlock(MethodBlock),
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum ReportingLogicType {
+    Focus,
```
Comment (amousset) [inline language/src/technique.rs:208]: what does the "focus" type does?

Comment (peckpeck) [reply language/src/technique.rs:208]: Ask the author :-)
I means : "look and the value element to extract the reporting of a single sub item"

Comment (amousset) [reply language/src/technique.rs:208]: so this one needs to be put on one of the states inside of the technique?

Comment (peckpeck) [reply language/src/technique.rs:208]: This one is the technique editor representation of this information, but you are right, it should be a metadata of a state inside the block in the language

Comment (amousset) [reply language/src/technique.rs:208]: LGTM with a comment explaining the focus case requires additional stuff

Comment (VinceMacBuche) [reply language/src/technique.rs:208]: I think it should be represented differently.

Focus should always have a "value" parameter and the other should never have one. I think their type should represent that

Comment (VinceMacBuche) [reply language/src/technique.rs:208]: More like 
enum ReportingLogic {
    Focus { value : String },
   Worst,
  Sum
}

Comment (peckpeck) [reply language/src/technique.rs:208]: indeed

Comment (peckpeck) [reply language/src/technique.rs:208]: but this woudn't match the json structure

Comment (VinceMacBuche) [reply language/src/technique.rs:208]: I think it should match the semantic of the program we want not the json structure, we don't want a value for Sum or Worst,and we always want one for Focus (where missing Value is an error and not optionnal)

**Comment by peckpeck** [comment]
done

**Review by VinceMacBuche — APPROVED** [review]

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**